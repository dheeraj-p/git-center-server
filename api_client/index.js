import fetch from 'isomorphic-unfetch';

export async function httpPOST(url, body) {
  const response = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify(body),
  });
  return response.json();
}

export async function httpGET(url) {
  const response = await fetch(url);
  return response.json();
}

export function createRepository(repositoryName, owner, baseURL = '') {
  return httpPOST(`${baseURL}/api/repositories`, { repositoryName, owner });
}

export function addSSHKey(sshKey, baseURL = '') {
  return httpPOST(`${baseURL}/api/ssh`, { sshKey });
}

export function getGitObject(repository, type, path, baseURL = '') {
  const repositoryURL = [repository, type, ...path].join('/');
  return httpGET(`${baseURL}/api/repository/${repositoryURL}`);
}

export function createUser(name, username, password, baseURL = '') {
  return httpPOST(`${baseURL}/api/user`, { name, username, password });
}

export function login(username, password, baseURL = '') {
  return httpPOST(`${baseURL}/api/login`, { username, password });
}

export function validateSession(username, token, baseURL = '') {
  return httpPOST(`${baseURL}/api/validateSession`, { username, token });
}

export function getUserRepositories(username, baseURL = '') {
  return httpPOST(`${baseURL}/api/userRepositories`, { username });
}
