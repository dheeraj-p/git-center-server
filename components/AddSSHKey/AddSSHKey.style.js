import css from 'styled-jsx/css';
import colors from '../../styles/colors';

const AddSHHKeyStyles = css`
  .add-ssh {
    border: 1px solid black;
    width: 100%;
    padding: 15px;
    margin: 15px;
  }

  .input-box {
    display: flex;
  }
`;

export default AddSHHKeyStyles;
