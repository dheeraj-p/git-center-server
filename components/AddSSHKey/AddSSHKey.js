import { useState } from 'react';
import * as APIClient from '../../api_client';

export default function AddSSHKey() {
  const [sshKey, setSSHKey] = useState('');
  const [messageState, setMessageState] = useState({value: '', class: ''});
  const [buttonDisable, setButtonDisable] = useState(false);

  const onSSHKeyChange = e => {
    setSSHKey(e.target.value);
  };

  const onSSHKeyAdd = async () => {
    setButtonDisable(true);
    setMessageState({value: 'Adding Key...', class: 'info'});
    const res = await APIClient.addSSHKey(sshKey);
    setButtonDisable(false);
    if(res.error) {
      setMessageState({value: 'Could not add SSH key!', class: 'danger'});  
      return
    }
    setMessageState({value: 'Your SSH key has been added!', class: 'success'});
    setSSHKey('');
  };

  return (
    <div className='card mt-2'>
      <div className='card-body'>
        <h5 className='card-title'>Add SSH Key</h5>
        <p className='card-text text-muted'>
          Paste your public SSH key to access repositories.
        </p>
        <p className={`m-0 mb-1 text-${messageState.class}`} id="message">
          <small>{messageState.value}</small>
        </p>
        <div className='row no-gutters'>
          <div className='col-12 col-md-9 mb-2 mb-md-0'>
            <textarea
              className='form-control'
              placeholder='Paste you public key here'
              onChange={onSSHKeyChange}
              value={sshKey}
            />
          </div>
          <div className='col-12 col-md-3 align-self-end'>
            <button
              type='button'
              className='btn btn-success w-100 ml-md-2'
              onClick={onSSHKeyAdd}
              disabled={buttonDisable}
            >
              Add Key
            </button>
          </div>
        </div>
      </div>
    </div>
  );
}
