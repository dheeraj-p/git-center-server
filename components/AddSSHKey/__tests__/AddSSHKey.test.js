import { shallow } from 'enzyme';
import AddSSHKey from '../AddSSHKey';
import * as APIClient from '../../../api_client';

jest.mock('../../../api_client');

describe('AddSSHKey', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<AddSSHKey />);
  });

  const simulateOnChangeOnInputField = newText => {
    wrapper.find('textarea').simulate('change', {
      target: {
        value: newText
      }
    });
  };

  it('should render properly', () => {
    expect(wrapper.debug()).toMatchSnapshot();
  });

  it('should set the ssh key on change the ssh key text-area', () => {
    simulateOnChangeOnInputField('test-key');
    expect(wrapper.find('textarea').prop('value')).toEqual('test-key');
  });

  it('should make an API call /api/ssh on click create', async () => {
    simulateOnChangeOnInputField('test-key');
    wrapper.find('button').simulate('click');
    expect(APIClient.addSSHKey).toHaveBeenCalledWith('test-key');
  });

  it('should show ssh key success message', async () => {
    simulateOnChangeOnInputField('test-key');
    wrapper.find('button').simulate('click');

    await wrapper.update();

    expect(wrapper.find('#message').text()).toBe(
      'Your SSH key has been added!'
    );
  });
});
