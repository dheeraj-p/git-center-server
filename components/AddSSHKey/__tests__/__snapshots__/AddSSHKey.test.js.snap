// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddSSHKey should render properly 1`] = `
"<div className=\\"card mt-2\\">
  <div className=\\"card-body\\">
    <h5 className=\\"card-title\\">
      Add SSH Key
    </h5>
    <p className=\\"card-text text-muted\\">
      Paste you public SSH key to access repositories.
    </p>
    <div className=\\"row no-gutters\\">
      <div className=\\"col-12 col-md-9 mb-2 mb-md-0\\">
        <textarea className=\\"form-control\\" placeholder=\\"Paste you public key here\\" />
      </div>
      <div className=\\"col-12 col-md-3 align-self-end\\">
        <button type=\\"button\\" className=\\"btn btn-primary w-100 ml-md-2\\">
          Add Key
        </button>
      </div>
    </div>
  </div>
</div>"
`;
